keps/sig-storage/2857-runtime-assisted-pv-mounts/README.md

```diff
@@ -0,0 +1,1387 @@
+multiple writers like: ReadWriteOnce (which allows multiple pods in the same
+node to mount a PV in read/write) and ReadWriteMany should be used with
+caution depending on the isolation goals. Multiple writer access modes (within
+the same or distinct nodes) allow a compromised/malicious pod to write data
+into a shared volume that may affect: [1] other pods reading the data through
+corrupt/fuzzed data and [2] other writers through denial of service attacks like
+inode exhaustion on the volume. As a result, goals around isolation between pods
+may be compromised. A cluster admin can configure a webhook/OPA policies to
+restrict the set of access modes that can be specified on PVCs that are referred
+by pods associated with a microvm runtime.
+
+A pod typically maps to the isolated sandbox environment in the context of
+microvm runtimes. Individual containers in the pod within the sandbox are not
+expected to be isolated with the same guarantees that exist across pods. To
+align with these isolation goals, restrictions around the ability of multiple
+containers within a pod mounting the same PV should not be necessary and
+considered beyond the scope of this KEP.
+
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Enhancements to existing APIs
+
+As summarized in the Proposal section above, coordination of mounting and
+management of the filesystem between a CSI plugin and a container runtime requires
+enhancements to multiple APIs and components of a Kubernetes cluster. This section
+delves into the details of each enhancement or addition.
+
+#### Enhancements in RuntimeClass
+
+New fields are necessary in RuntimeClass to indicate: [1] a handler/runtime can
+support mounting and management of file systems on PVs mounted by a pod and
+[2] specify a domain socket path which Kubelet can use to invoke the FileSystem
+Management APIs (described later) that will be handled by the container runtime.
+
+```
+type RuntimeClass struct {
+	metav1.TypeMeta `json:",inline"`
+  ...
+  // SupportsFSMounts indicates whether the Handler supports mounting of
+  // FileSystem for persistent volumes referred by the pod
+  // Default False
+  SupportsFSMounts bool `json:"supportsFSMounts,omitempty" protobuf:"bytes,5,opt,name=supportsFSMounts"`
```

Comment:
API reviewers will know this better, but I suspect this might have to be a pointer to handle cases where the client doesn't support the field (set vs. not set).